http-server-netty/src/main/java/io/micronaut/http/server/netty/RoutingInBoundHandler.java

```diff
@@ -328,6 +333,9 @@ private void exceptionCaughtInternal(ChannelHandlerContext ctx,
                 Class<?> javaReturnType = errorRoute.getReturnType().getType();
                 boolean isFuture = CompletionStage.class.isAssignableFrom(javaReturnType);
                 boolean isReactiveReturnType = Publishers.isConvertibleToPublisher(javaReturnType) || isFuture;
+                boolean isKotlinSuspendingFunction = isKotlinSuspendingFunction(methodBasedRoute.getTargetMethod());
```

Comment:
This bit concerns me as the usage of `getTargetMethod()` is discouraged as it results in reflection metadata being initialized in the controller class which we are trying to avoid in Micronaut.

Is there another way to check whether it is a suspending function? Or is reflection required to do so?

Reply:
I got rid of Java/Kotlin Reflection API usage and am relying now on `ExecutableMethod` metadata. There is a difference, since I'm assuming that any method with `Continuation` argument as the last argument is a `suspend` function. In theory we could a regular function which has this property, but in reality I see no reason to have such a function in HTTP controller.